cobigen/cobigen-core/src/test/java/com/devonfw/cobigen/unittest/config/upgrade/TemplateSetUpgraderTest.java

```diff
@@ -33,151 +24,135 @@
+    int OldTemplatesFileCount = oldTemplatesPath.toFile().list().length;
+    Set<String> OldPathFilesSet = new HashSet<>(Arrays.asList(oldTemplatesPath.toFile().list()));
+
+    TemplateSetUpgrader templateSetUpgrader = new TemplateSetUpgrader();
+    templateSetUpgrader.upgradeTemplatesToTemplateSets(this.templateLocation);
+
+    Path backupPath = this.templateLocation.getParent().resolve("backup")
+        .resolve(ConfigurationConstants.TEMPLATES_FOLDER).resolve(ConfigurationConstants.COBIGEN_TEMPLATES)
+        .resolve(ConfigurationConstants.TEMPLATE_RESOURCE_FOLDER);
+    Set<String> backupPathFilesSet = new HashSet<>(Arrays.asList(backupPath.toFile().list()));
+    Path newTemplatesPath = this.templateLocation.getParent().resolve(ConfigurationConstants.TEMPLATE_SETS_FOLDER)
+        .resolve(ConfigurationConstants.ADAPTED_FOLDER);
+    Set<String> NewPathFilesSet = new HashSet<>(Arrays.asList(newTemplatesPath.toFile().list()));
+
+    assertEquals(OldTemplatesFileCount - 1, NewPathFilesSet.size());
+    assertEquals(OldTemplatesFileCount, backupPathFilesSet.size());
+
+    for (String s : OldPathFilesSet) {
+      if (!s.equals("context.xml")) {
+        assertTrue(NewPathFilesSet.contains(s));
+        NewPathFilesSet.remove(s);
+      }
+      assertTrue(backupPathFilesSet.contains(s));
+      backupPathFilesSet.remove(s);
+
+    }
+    assertEquals(NewPathFilesSet.size(), 0);
+    assertEquals(backupPathFilesSet.size(), 0);
+  }
+
+  /**
+   * Tests the correct location for the created context.xml and if the files and whether the file corresponds to the
+   * v3.0 schema
+   *
+   * @throws Exception
+   */
+  @Test
+  public void testTemplateSetUpgradeContextSplit() throws Exception {
+
+    Path contextLocation = this.templateLocation.resolve(ConfigurationConstants.COBIGEN_TEMPLATES)
+        .resolve(ConfigurationConstants.TEMPLATE_RESOURCE_FOLDER);
+    ContextConfigurationUpgrader upgrader = new ContextConfigurationUpgrader();
+    upgrader.upgradeConfigurationToLatestVersion(contextLocation, BackupPolicy.ENFORCE_BACKUP);
+
+    Path newTemplatesPath = this.templateLocation.getParent().resolve(ConfigurationConstants.TEMPLATE_SETS_FOLDER);
+    newTemplatesPath = newTemplatesPath.resolve(ConfigurationConstants.ADAPTED_FOLDER);
+
+    SchemaFactory schemaFactory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
+    try (InputStream schemaStream = getClass().getResourceAsStream("/schema/v3.0/contextConfiguration.xsd")) {
+      StreamSource schemaSourceStream = new StreamSource(schemaStream);
+      Schema schema = schemaFactory.newSchema(schemaSourceStream);
+      Validator validator = schema.newValidator();
+
+      for (String s : newTemplatesPath.toFile().list()) {
+        Path newContextPath = newTemplatesPath.resolve(s + "/" + "src/main/resources");
+        newContextPath = newContextPath.resolve("context.xml");
+        assertTrue(newContextPath.toFile().exists());
```

Comment:
Use assertThat(...).exists();

Check for example velocityGenerationTest for reference